test/server/cards/01_SOR/units/HomesteadMilitia.spec.js

```diff
@@ -0,0 +1,36 @@
+describe('Homestead Militia', function () {
+    integration(function () {
+        describe('Homestead Militia\'s ability', function () {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['homestead-militia'],
+                        resources: ['distant-patroller', 'distant-patroller', 'distant-patroller', 'distant-patroller', 'distant-patroller']
+                    },
+                    player2: {
+                        groundArena: ['rugged-survivors', 'cargo-juggernaut']
+                    }
+                });
+            });
+
+            it('should not gain Sentinel with 6 or more resources', function () {
+                this.player1.pass();
+
+                this.player2.clickCard(this.ruggedSurvivors);
+                // no sentinel, I can attack base
+                expect(this.player2).toBeAbleToSelectExactly([this.p1Base, this.homesteadMilitia]);
+                this.player2.clickCard(this.p1Base);
+                expect(this.p1Base.damage).toBe(3);
+
+                this.player1.resources.push('distant-patroller');
```

Comment:
Let's not do this this way, I'm not 100% sure it will work right in all cases since adding cards using a string really only works well in `setupTest`. Using strings after that can be problematic because either (a) the test method may not know how to create a card from the string name, and (b) even if it does, that card probably doesn't exist in the player's list of cards so some checks may not work right.

A thing you could do is add like a wampa to hand and then go `player1.moveCard(this.wampa, 'resources');`. I _think_ that should work.

Ditto for other places in the PR this is done.

Reply:
moveCard seems to not work. Maybe I can just make 2 describes ?
> AssertionError: Contract assertion failure: Attempting to find pile 'resources', but it does not exist for player1. Use createAdditionalPile() to add new pile types.

Reply:
Ah sorry, I had a typo - it should be `resource`, not `resources`. Otherwise I think it is good to keep this all in one test case so we can have some extra validation that dynamic gain / lose keyword is working right.